RecoPPS/Local/plugins/CTPPSLocalTrackLiteReAligner.cc

```diff
@@ -0,0 +1,107 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/CTPPSReco/interface/CTPPSLocalTrackLite.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSLocalTrackLiteFwd.h"
+
+#include "CondFormats/AlignmentRecord/interface/RPRealAlignmentRecord.h"
+#include "CondFormats/PPSObjects/interface/CTPPSRPAlignmentCorrectionsData.h"
+
+//----------------------------------------------------------------------------------------------------
+
+class CTPPSLocalTrackLiteReAligner : public edm::stream::EDProducer<> {
+public:
+  explicit CTPPSLocalTrackLiteReAligner(const edm::ParameterSet &);
+
+  void produce(edm::Event &, const edm::EventSetup &) override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &);
+
+private:
+  edm::EDGetTokenT<CTPPSLocalTrackLiteCollection> inputTrackToken_;
+
+  std::string alignmentLabel_;
+
+  std::string outputTrackTag_;
+};
+
+//----------------------------------------------------------------------------------------------------
+
+CTPPSLocalTrackLiteReAligner::CTPPSLocalTrackLiteReAligner(const edm::ParameterSet &iConfig)
+    : inputTrackToken_(consumes<CTPPSLocalTrackLiteCollection>(iConfig.getParameter<edm::InputTag>("inputTrackTag"))),
+      alignmentLabel_(iConfig.getParameter<std::string>("alignmentLabel")),
+      outputTrackTag_(iConfig.getParameter<std::string>("outputTrackTag")) {
+  produces<CTPPSLocalTrackLiteCollection>(outputTrackTag_);
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void CTPPSLocalTrackLiteReAligner::fillDescriptions(edm::ConfigurationDescriptions &descr) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>("inputTrackTag", edm::InputTag("ctppsLocalTrackLiteProducer"))
+      ->setComment("tag of the input CTPPSLocalTrackLiteCollection");
+
+  desc.add<std::string>("alignmentLabel", "")->setComment("label of the alignment data");
+
+  desc.add<std::string>("outputTrackTag", "")->setComment("tag of the output CTPPSLocalTrackLiteCollection");
+
+  descr.add("ctppsLocalTrackLiteReAligner", desc);
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void CTPPSLocalTrackLiteReAligner::produce(edm::Event &iEvent, const edm::EventSetup &iSetup) {
+  // get alignment correction
+  edm::ESHandle<CTPPSRPAlignmentCorrectionsData> hAlignment;
+  iSetup.get<RPRealAlignmentRecord>().get(alignmentLabel_, hAlignment);
+
+  // get input
+  edm::Handle<CTPPSLocalTrackLiteCollection> hInputTracks;
+  iEvent.getByToken(inputTrackToken_, hInputTracks);
+
+  // prepare output
+  auto output = std::make_unique<CTPPSLocalTrackLiteCollection>();
+
+  // apply alignment correction
+  for (const auto &tr : *hInputTracks) {
+    auto it = hAlignment->getRPMap().find(tr.rpId());
+    if (it == hAlignment->getRPMap().end()) {
+      edm::LogError("CTPPSLocalTrackLiteReAligner::produce")
```

Comment:
the module class name and instance name are already a part of the LogError formatting.
The category should be more generic and if needed indicative of the type of issue.
So a CTPPS can be a good enough category.
```suggestion
      edm::LogError("CTPPS")
```

Also, IIRC, the :: are not really acceptable in the category name.

More specifically to the problem here: how often is this expected and is this worth to continue the job?
or if the issue is acceptable, then LogInfo is more appropriate.

Reply:
For most PPS modules we have used the convention that category was the class::method. I liked that the method is indicated, often telling what kind of problems has occurred.

We can change the convention (in steps). I have no big problem with "CTPPS". Still as we are (slowly) migrating towards "PPS" wouldn't it make sense to start using "PPS" right away?

Reply:
For a properly configured job, no error should appear. Conversely, an error appearing means that the results can be seriously wrong and that the configuration should be improved.

Reply:
PPS is fine.
This is actually a good point to bring up why is this new module code called "CTPPS"?

https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger#CMSSW_Guidelines_for_Messages_an
LogError/Warnings are expected to have more clear categories.
 For Info and Debug, category names can include the function that is being performed. 
name::member appear to be acceptable for lower level method calls, which is not the case here. IIRC, the standard producer message formatting already includes info if the call is from constructor or beginRun/Lumi transition.

Reply:
I've changed the message category to PPS.

Reply:
For the module name: originally, I found adding a "PPS" module into a directory full of "CTPPS" modules inconsistent. On the other, we agreed to call new stuff PPS. Therefore I've renamed the new module to "PPS".